### PR TITLE
Improve get user authorized organization endpoint

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
- Improve get user authorized organization endpoint

## Approach
- At the moment, we identify a given organization is an authorized organization if a user association with the authenticated user for that given organization exist. We can improve that logic further to check the associated user has a role associated with him to indicate it is an authorized organization.

## Related Issue
- https://github.com/wso2/product-is/issues/18727

##  Merge After
- https://github.com/wso2/identity-organization-management-core/pull/123

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/factory/ApplicationManagementOSGIServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package org.wso2.carbon.identity.api.user.organization.common.factory;
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/factory/ApplicationManagementOSGIServiceFactory.java:1]: missing license header

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/resources/user.organization.yaml [hunk 2]
```diff
@@ -163,6 +164,14 @@ components:
       schema:
         type: boolean
         default: false
+    applicationNameQueryParam:
+        in: query
+        name: applicationName
+        required: false
+        description:
+            Filter organizations that user has access due to roles of the particular application
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/resources/user.organization.yaml:172]: there is already a filter query param defined for this API. can't we reuse it?

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/factory/ApplicationManagementOSGIServiceFactory.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/factory/ApplicationManagementOSGIServiceFactory.java:19]: Fixed

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/resources/user.organization.yaml (hunk 2 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/resources/user.organization.yaml]: Agreed on using a separate param

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/resources/user.organization.yaml [hunk 3]
```diff
@@ -163,6 +164,14 @@ components:
       schema:
         type: boolean
         default: false
+    authorizedAppNameQueryParam:
+        in: query
+        name: authorizedAppName
+        required: false
+        description:
+          Retrieves authorized organizations for a user with a role in the given application.
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/resources/user.organization.yaml:172]: Retrieves the organizations that are authorized for the user through role bound to the application.
```suggestion
          Retrieves authorized organizations for a user with a role in the given application.
```

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/factory/ApplicationManagementOSGIServiceFactory.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/factory/ApplicationManagementOSGIServiceFactory.java:2]:
```suggestion
 * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
```

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/UserOrganizationServiceHolder.java [hunk 5]
```diff
@@ -71,4 +73,25 @@ public static void setOrganizationManagementService(
...
+     *
+     * @return Application management service.
+     */
+    public static ApplicationManagementService getApplicationManagementService() {
+
+        return applicationManagementService;
+    }
+
+    /**
+     * Set Application management OSGi service.
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.api.user.organization.common/src/main/java/org/wso2/carbon/identity/api/user/organization/common/UserOrganizationServiceHolder.java:88]:
```suggestion
     * Set application management OSGi service.
```

**Review by dewniMW — APPROVED** [review]

**Pull request merged**